wsd/DocumentBroker.cpp

```diff
@@ -3373,11 +3367,13 @@ std::size_t DocumentBroker::countActiveSessions() const
 void DocumentBroker::setModified(const bool value)
 {
 #if !MOBILEAPP
+    // Flag the document as modified in the admin console.
+    Admin::instance().modificationAlert(_docKey, getPid(), value);
     if (value)
     {
-        // Flag the document as modified in the admin console.
-        // But only flag it as unmodified when we do upload it.
-        Admin::instance().modificationAlert(_docKey, getPid(), value);
+        // Flag the document uploaded status to false in admin panel,
+        // as the document got modified
+        Admin::instance().uploadedAlert(_docKey, getPid(), !value);
```

Comment:
We are setting the uploaded state _only_ when the document is modified. But we should consider the race condition that exists between uploading and getting the `modifiedstatus=false` from Core.

Since we can't control when we will get the `modifiedstatus`, we need to simply update the uploaded state when we do, in a generic way.

That is, we shouldn't call `uploadedAlert` only when modifiedstatus=true, instead, we should call this and pass the actual state of the upload, which is given by `needToUploadToStorage()`. But then there might be an upload in flight already, so we need to take that into account too.

Calling this _unconditionally_ here should work:
```
    Admin::instance().uploadedAlert(
        _docKey, getPid(), !isAsyncUploading() && needToUploadToStorage() == NeedToUpload::No);
```

Reply:
Thanks, now the whole picture got cleared to me why my approach was wrong

Reply:
I have one question wouldn't it be confusing when admin sees Modified=No but Uploaded=Yes
as Uploaded is true because it updates the status based on `needToUploadToStorage() -> isStorageOutdated()`